[WFLY-18240]: org.apache.activemq.artemis is required as an explicit …

ehsavoie opened the pull request:
…dependency for some deployments

 * Creating a org.apache.activemq.artemis.client module
 * Adding a org.apache.activemq.artemis.service-extensions module
 * Making the org.apache.activemq.artemis optional
 * Adding a ActiveMQBroker and ActiveMQBrokerImpl classes to wrap the ActiveMQServer so that we don't have direct references to it

Issue: https://issues.redhat.com/browse/WFLY-18240

bstansberry commented on messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQBroker.java:
```diff
@@ -0,0 +1,50 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
Please use the short copyright/license text.

bstansberry commented on messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQBrokerImpl.java:
```diff
@@ -0,0 +1,94 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
Please use the short copyright/license text.

bstansberry commented on messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/jms/bridge/JMSBridgeFactory.java:
```diff
@@ -0,0 +1,118 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
Please use the short copyright/license text.

bstansberry commented on messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQBroker.java:
```diff
@@ -0,0 +1,50 @@
+import org.apache.activemq.artemis.core.security.SecurityAuth;
+
+/**
+ *
```
This needs class javadoc, e.g.

"Wrapper interface to expose methods from {@link org.apache.activemq.artemis.core.server.ActiveMQServer} and {@link org.apache.activemq.artemis.core.server.management.ManagementService} that are needed by a {@code /subsystem=messaging-activemq/server=*} resource or its children."

bstansberry commented on messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQBroker.java:
```diff
@@ -0,0 +1,50 @@
+ */
+public interface ActiveMQBroker {
+
+    Object getDelegate();
```
This one needs javadoc too.

I assume the other methods are just copied from ActiveMQServer or ManagementService, so could be doced with a simple "@see {@link .....#..."

ehsavoie replied:
I wrapped also call to classes present in the artemis server jar via the ActiveMQServer

bstansberry commented on ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/apache/activemq/artemis/main/module.xml:
```diff
@@ -36,16 +36,11 @@
     </properties>
```
I guess not right now, as you export the client stuff here, which makes this module still usable as a deployment dependency. Whenever you drop that export would be the logically time to change jboss.api to 'private'.

ehsavoie replied:
So I'm moving this to private so that the comment makes sense

ehsavoie committed:
[WFLY-18240]: org.apache.activemq.artemis is required as an explicit dependency for some deployments

 * Creating a org.apache.activemq.artemis.client module
 * Adding a org.apache.activemq.artemis.service-extensions module
 * Making the journal module optional
 * Making the org.apache.activemq.artemis optional
 * Adding a ActiveMQBroker and ActiveMQBrokerImpl classes to wrap the
   ActiveMQServer so that we don't have direct references to it
 * Making the "org.apache.activemq.artemis" module not useable from
   deployments which should use the "org.apache.activemq.artemis.client" module.

Issue: https://issues.redhat.com/browse/WFLY-18240
Signed-off-by: Emmanuel Hugonnet <ehugonne@redhat.com>

ehsavoie committed:
[WFLY-18756]: The recovery connection factory ignores use-topology-for-load-balancing.

 * When the recovery connection factory is created in
   ActiveMQResourceAdapter it ignores the use-topology-for-load-balancing parameter.

Issue: https://issues.redhat.com/browse/WFLY-18756

Signed-off-by: Emmanuel Hugonnet <ehugonne@redhat.com>

bstansberry commented:
Thanks @ehsavoie
